### PR TITLE
Rtrt email in article

### PR DESCRIPTION
**Pull request by gtrufitt**
Allow embedding of the inline email into composer articles. Embed code looks like:

`<iframe src="/email/form/article" data-form-title="Want stories like this in your inbox?" data-form-description="Subscribe to the Guardian Today and get the biggest headlines delivered to you each morning" scrolling="no" seamless frameborder="0" class="iframed--overflow-hidden email-sub__iframe js-email-sub__iframe--article"></iframe>`

image

image

I have update the email routes to allow a form type, restricted initially to `footer` and `article`. Passing options/settings from the embedded iframe to the form below is done with data attributes.

### CODE REVIEW

**Comment by janua** [comment]
LGTM :+1:

File: applications/app/controllers/EmailController.scala [hunk 1]
```diff
@@ -66,8 +66,8 @@ object EmailController extends Controller with ExecutionContexts with Logging {
     Cached(60)(Ok(views.html.emailLanding(emailLandingPage)))
   }
 
-  def renderForm() = Action { implicit request =>
-    Cached(60)(Ok(views.html.emailFragment(emailLandingPage)))
+  def renderForm(formType: String) = Action { implicit request =>
+    Cached(60)(Ok(views.html.emailFragment(emailLandingPage, None, formType)))
```
Comment (NathanielBennett) [inline applications/app/controllers/EmailController.scala]: Think you can omit passing "None"

File: applications/conf/routes [hunk 2]
```diff
@@ -22,11 +22,11 @@ GET            /crosswords/search
...
-GET        /email/form                                                          controllers.EmailController.renderForm()
-GET        /email/:result                                                       controllers.EmailController.subscriptionResult(result: String)
-POST       /email                                                               controllers.EmailController.submit()
-OPTIONS    /email                                                               controllers.EmailController.options()
+GET            /email                                                           controllers.EmailController.renderPage()
+GET            /email/form/$formType<footer|article>                            controllers.EmailController.renderForm(formType: String)
+GET            /email/:result                                                   controllers.EmailController.subscriptionResult(result: String)
+POST           /email                                                           controllers.EmailController.submit()
+OPTIONS        /email                                                           controllers.EmailController.options()
```
Comment (NathanielBennett) [inline applications/conf/routes]: Mebbe get this so the whitespace changes don't get picked up at changes

Comment (gtrufitt) [inline applications/conf/routes]: Yeah, IntelliJ winds me up

**Comment by NathanielBennett** [comment]
:+1: Otherwise

File: applications/app/controllers/EmailController.scala (hunk 1 above)
Comment (janua) [inline applications/app/controllers/EmailController.scala]: You can only if you name the parameters.

Comment (gtrufitt) [inline applications/app/controllers/EmailController.scala]: Should they be named or is this ok?

**Comment by OliverJAsh** [comment]
Nice work :+1:

**Pull request merged**